apps/dotcms-ui/src/app/portlets/shared/dot-content-types-listing/components/dot-add-to-menu/dot-add-to-menu.component.ts

```diff
@@ -0,0 +1,145 @@
+import { Observable, Subject } from 'rxjs';
+import {
+    Component,
+    Input,
+    Output,
+    EventEmitter,
+    OnInit,
+    OnDestroy,
+    ViewChild,
+    ElementRef
+} from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { DotMessageService } from '@services/dot-message/dot-messages.service';
+import { DotDialogActions } from '@components/dot-dialog/dot-dialog.component';
+import { DotMenuService } from '@dotcms/app/api/services/dot-menu.service';
+import { switchMap, take, takeUntil, tap } from 'rxjs/operators';
+import { DotMenu } from '@dotcms/app/shared/models/navigation';
+import { DotCMSContentType } from '@dotcms/dotcms-models';
+import {
+    DotAddToMenuService,
+    DotCreateCustomTool
+} from '@dotcms/app/api/services/add-to-menu/add-to-menu.service';
+
+@Component({
+    selector: 'dot-add-to-menu',
+    templateUrl: 'dot-add-to-menu.component.html'
+})
+export class DotAddToMenuComponent implements OnInit, OnDestroy {
+    form: FormGroup;
+    menu$: Observable<DotMenu[]>;
+    placeholder = '';
+    dialogShow = false;
+    dialogActions: DotDialogActions;
+
+    @Input() contentType: DotCMSContentType;
+    @Output() cancel = new EventEmitter<boolean>();
+
+    @ViewChild('formEl', { static: true }) formEl: HTMLFormElement;
+    @ViewChild('titleName', { static: true }) titleName: ElementRef;
+
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+
+    constructor(
+        public fb: FormBuilder,
+        private dotMessageService: DotMessageService,
+        private dotMenuService: DotMenuService,
+        private dotAddToMenuService: DotAddToMenuService
+    ) {}
+
+    ngOnInit() {
+        this.initForm();
+        this.setDialogConfig(this.form);
+
+        this.menu$ = this.dotMenuService.loadMenu().pipe(
+            take(1),
+            tap((menu: DotMenu[]) => {
+                this.form.patchValue({
+                    menuOption: menu[0].id
+                });
+                this.titleName.nativeElement.focus();
+            })
+        );
+    }
+
+    ngOnDestroy(): void {
+        this.destroy$.next(true);
+        this.destroy$.complete();
+    }
+
+    /**
+     * Close dialog modal and reset form
+     * @memberof DotAddToBundleComponent
+     */
+    close(): void {
+        this.cancel.emit(true);
+    }
+
+    /**
+     * Add to bundle if form is valid
+     * @memberof DotAddToBundleComponent
+     */
+    submit(): void {
+        if (this.form.valid) {
+            const cleanPorletId = this.dotAddToMenuService.cleanUpPorletId(
+                this.form.get('title').value
+            );
```

Comment:
Why don't clean up this inside the server?

Reply:
this is the way it's being done on the ROLES & TOOLS portlet (on the JSP via JS)

Reply:
I mean... you are cleaning the string here but you can do all that internally in the service.